### ISSUE TITLE
HW1 질문

### ISSUE THREAD

**Issue by hashnut** [snu-sf-class/swpp201901#21]
PUT 기능에서 keyerror가 나는 경우에 대해 여쭤보고 싶습니다.

새로운 meeting 시간을 생성할 때 겹치는 시간을 처리할 때는 

`serializer.validated_data["sinceWhen"] 같은 식으로 확인이 가능한 것 같습니다.

그런데 PUT 에서,

`def perform_update(self, serializer):`

함수를 override 하는 과정에서, 
`currentId = serializer.validated_data["id"]`
같은 식으로 ID를 조회하려하면 `keyerror`가 발생합니다.

`serializer`를 뜯어보면 `MeetingSerializer Object<1>` 같은 식으로 표현이 되어 있는데 왜 조회할 수 없는 건지 궁금합니다.

PUT 시 현재 업데이트 하고 있는 객체의 번호는 어떤식으로 가져올 수 있을까요?

**Comment by ldm2468** [comment]
저 에러가 뜨는 이유는 `serializer.validated_data`에 바뀐 값만 있어서 그런 것 같습니다.
일단 `perform_update` 함수에서 meeting시간을 validation하는 것 자체가 antipattern이기 때문에 validation 코드를 serializer 내부로 옮기는게 좋을 것 같습니다.
Serializer의 `validate(self, data)` 함수에선 self.instance.id로 업데이트 중인 개체의 id를 조회할 수 있습니다.
(https://www.django-rest-framework.org/api-guide/serializers/#object-level-validation 참고)

**Comment by ldm2468** [comment]
급하다면 `serializer.instance.id`로 `perform_update`내에서도 현재 개체의 id를 읽을 수 있을지도 모르겠네요. (테스트해보진 않음)

**Comment by hashnut** [comment]
@ldm2468 
답변 감사합니다! 그런데 serializer 내부에 validate(self, data)함수를 만들면 자동으로 체크를 해 주나요? 결국 구현은 못했지만 궁금해서 여쭤봅니다!

**Comment by ldm2468** [comment]
네, `validate(self, data)` 함수를 만들면 `serializer.is_valid()`를 부를때 자동으로 체크가 됩니다!

**Comment by hashnut** [comment]
답변 감사합니다 :)
